### PR TITLE
add initial android support for the firebase BCNY workflow

### PR DESCRIPTION
**Pull request by hyp**
the android build is based on the original google android build.

I omitted the NuGet packaging for android for now, as I want to verify the currently built libraries first to see if I need more libraries like abseil on Windows.

### CODE REVIEW

**Comment by hyp** [comment]
example: 
https://github.com/thebrowsercompany/firebase-cpp-sdk/releases/tag/20240517.0

**Review by tristanlabelle — APPROVED** [review]
Looks good!

File: .github/workflows/bcny-firebase.yml [hunk 1]
```diff
@@ -348,3 +348,137 @@ jobs:
...
+          cmake_build $arch "analytics" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_ANALYTICS=ON"
+          cmake_build $arch "database" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_DATABASE=ON"
+          cmake_build $arch "gma" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_GMA=ON"
+          cmake_build $arch "dynamic_links" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_DYNAMIC_LINKS=ON"
+          cmake_build $arch "installations" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_INSTALLATIONS=ON"
+          cmake_build $arch "app_check" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_APP_CHECK=ON"
+          cmake_build $arch "firestore" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_FIRESTORE=ON"
+          cmake_build $arch "auth" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_AUTH=ON"
+          cmake_build $arch "functions" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_FUNCTIONS=ON"
+          cmake_build $arch "remote_config" "-DFIREBASE_CPP_USE_PRIOR_GRADLE_BUILD=ON -DFIREBASE_INCLUDE_REMOTE_CONFIG=ON"
```
Comment (compnerd) [inline .github/workflows/bcny-firebase.yml:435]: This feels like its building more than we do on Windows, but I think that getting it built is more important than optimizing right now.

File: .github/workflows/bcny-firebase.yml [hunk 2]
```diff
@@ -348,3 +348,137 @@ jobs:
...
+      - name: Install (manual)
+        run: |
+          echo "Copying static libraries ..."
+          source="${{ github.workspace }}/SourceCache/firebase-cpp-sdk"
+          destination_dir=${{ github.workspace }}/BuildRoot/Library/firebase/usr/libs/android/${{ matrix.arch }}
+          mkdir -p $destination_dir
+          while IFS= read -r file; do
+            cp "$file" "$destination_dir"
+            echo "Copied: $file"
+          done < <(find $source -type f -name "*.a")
```
Comment (compnerd) [inline .github/workflows/bcny-firebase.yml:446]: Oh, we cannot use an install target from CMake? :(

**Review by compnerd — APPROVED** [review]
We can iterate on this once committed.

File: .github/workflows/bcny-firebase.yml (hunk 2 above)
Comment (hyp) [reply .github/workflows/bcny-firebase.yml:446]: no unfortunately

**Pull request merged**